Make WithSpriteBody and -Turret always use CancelCustomAnimation() after custom anim

reaperrr opened the pull request:
On bleed, custom animations that don't queue `CancelCustomAnimation()` will break the default sequence loop.

Closes #13742.

reaperrr committed:
Make WithSpriteBody.PlayCustomAnimation resume loop of Sequence

...after it finished playing.

pchote commented:
What uses PlayCustomAnimation right now?  Is there any reason to not make this work intuitively by default now?

reaperrr commented:
> What uses PlayCustomAnimation right now? Is there any reason to not make this work intuitively by default now?

Turns out the only case that doesn't CancelCustomAnimation() is (surprise, surprise...) With*Harvest*Animation, probably to avoid the default sequence overriding the fullness-dependant sequence.

That should be fixable, though, if that's even an issue in practice.

reaperrr committed:
Resume looping default WithSpriteTurret.Sequence after custom anim

It doesn't really make sense not to, and the only trait using PlayCustomAnimation previously did this manually anyway.

reaperrr committed:
WithHarvestAnimation style fixes

Make interface implementations explicit.

reaperrr commented:
Updated.

Requesting @forcecore to test this with #13737 (without the yaml work-around).

forcecore commented:
Thumper fix works without the walk around on this branch.

pchote commented:
> Turns out the only case that doesn't CancelCustomAnimation() is (surprise, surprise...) WithHarvestAnimation, probably to avoid the default sequence overriding the fullness-dependant sequence.

Are you sure about this? I would have assumed it was to let the harvesting animation loop if the animation length is less than the harvest cycle length.  If that's the case then this would be a good time/place to introduce a `PlayCustomAnimationRepeating` method.

pchote commented:
> I would have assumed it was to let the harvesting animation loop if the animation length is less than the harvest cycle length.

Either my assumption was wrong, or this was already broken on release/playtest/bleed.

pchote reviewed (approved):
Code changes look good to me, and no regressions spotted ingame.

abcdefg30 commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/2163f114e3ec5a6af967a351c77b8e57e2135c60...f2040025e6e079e9e9f407aa6c627128e546bdda)
